### DISCUSSION TITLE
How to enable web terminal exec into init containers

### DISCUSSION THREAD

**Discussion by helcim-adam-c** [argoproj/argo-cd#20511] [General]
Hello everyone,

I'm seeking assistance with enabling the Argo CD web terminal to execute commands within init containers. In our setup, we're utilizing Kubernetes' restartPolicy: Always feature to keep init containers active throughout the pod's lifecycle, as outlined in the [Kubernetes documentation on init containers](https://kubernetes.io/docs/concepts/workloads/pods/init-containers/).

Despite having admin privileges and the ability to exec into all pods, I encounter the following error when attempting to access init containers via the Argo CD web terminal:

`Terminal Connection Error: Internal error`

Is executing into init containers supported in Argo CD's web terminal? If so, are there specific configurations or permissions required to enable this functionality? 

Thank you for your assistance.

**Comment by wstarzak** [comment]
I do also have same problem, we have migrated init containers to fix gracefull shutdowns and unfortunately argocd exec is no longer available for this containers:

`argocd-server {"level":"error","msg":"unexpected closer error: websocket: close 1005 (no status)","time":"2025-12-03T10:34:29Z"}`                                
`argocd-server {"error":"websocket: close 1005 (no status)","level":"error","logger":"UnhandledError","msg":"Unhandled Error","time":"2025-12-03T10:34:29Z"} `
`argocd-server {"appNamespace":"eleonor","application":"eleonor-prod-main","container":"php","level":"warning","msg":"terminal container not found","namespace":"eleonor","podName":"eleonor-1f4dk","project":"eleonor","time":"2025-12-03T10:34:29Z","userName":"REDACTED"}`
